File: backend/internal/domain/family.go
package domain

import "time"

type Family struct {
	ID        string            `json:"id" firestore:"-"`
	Name      string            `json:"name" firestore:"name"`
	CreatedAt time.Time         `json:"createdAt" firestore:"createdAt"`
	CreatedBy string            `json:"createdBy" firestore:"createdBy"`
	MemberIDs []string          `json:"memberIds" firestore:"memberIds"`
	Members   map[string]Member `json:"members" firestore:"members"`
}

type Member struct {
	Role        MemberRole `json:"role" firestore:"role"`
	JoinedAt    time.Time  `json:"joinedAt" firestore:"joinedAt"`
	DisplayName string     `json:"displayName" firestore:"displayName"`
}

type MemberRole string

const (
	MemberRoleOwner     MemberRole = "owner"
	MemberRoleCaregiver MemberRole = "caregiver"
)

File: app/lib/features/family/domain/entities/family.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'family.freezed.dart';
part 'family.g.dart';

enum MemberRole { owner, caregiver }

@freezed
class Member with _$Member {
  const factory Member({
    required MemberRole role,
    required DateTime joinedAt,
    required String displayName,
  }) = _Member;

  factory Member.fromJson(Map<String, dynamic> json) => _$MemberFromJson(json);
}

@freezed
class Family with _$Family {
  const factory Family({
    required String id,
    required String name,
    required DateTime createdAt,
    required String createdBy,
    @Default([]) List<String> memberIds,
    @Default({}) Map<String, Member> members,
  }) = _Family;

  factory Family.fromJson(Map<String, dynamic> json) => _$FamilyFromJson(json);
}
